internal/wrappers/cgo/kuzzle/security.go

```diff
@@ -145,221 +131,332 @@ func kuzzle_security_destroy_role(r *C.role) {
+	err := (*kuzzle.Kuzzle)(k.instance).Security.DeleteCredentials(C.GoString(strategy), C.GoString(id), SetQueryOptions(o))
+	return goToCVoidResult(err)
+}
+
+//export kuzzle_security_create_profile
+func kuzzle_security_create_profile(k *C.kuzzle, id, body *C.char, o *C.query_options) *C.profile_result {
+	options := SetQueryOptions(o)
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateProfile(C.GoString(id), C.GoString(body), options)
+
+	return goToCProfileResult(k, res, err)
+}
+
+//export kuzzle_security_create_role
+func kuzzle_security_create_role(k *C.kuzzle, id, body *C.char, o *C.query_options) *C.role_result {
+	result := (*C.role_result)(C.calloc(1, C.sizeof_role_result))
+	options := SetQueryOptions(o)
+
+	role, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateRole(C.GoString(id), C.GoString(body), options)
+	if err != nil {
+		Set_role_result_error(result, err)
+		return result
 	}
 
-	res, err := cToGoUser(u).CreateWithCredentials(credentials, SetQueryOptions(o))
+	result.role = goToCRole(k, role, nil)
+
+	return result
+}
 
-	return goToCUserResult(u.kuzzle, res, err)
+//export kuzzle_security_create_user
+func kuzzle_security_create_user(k *C.kuzzle, body *C.char, o *C.query_options) *C.json_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateUser(C.GoString(body), SetQueryOptions(o))
+	return goToCJsonResult(res, err)
 }
 
-//export kuzzle_security_user_delete
-func kuzzle_security_user_delete(u *C.user, o *C.query_options) *C.string_result {
-	res, err := cToGoUser(u).Delete(SetQueryOptions(o))
-	return goToCStringResult(&res, err)
+//export kuzzle_security_create_credentials
+func kuzzle_security_create_credentials(k *C.kuzzle, cstrategy, id, body *C.char, o *C.query_options) *C.json_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateCredentials(C.GoString(cstrategy), C.GoString(id), C.GoString(body), SetQueryOptions(o))
+	return goToCJsonResult(res, err)
 }
 
-//export kuzzle_security_user_delete_credentials
-func kuzzle_security_user_delete_credentials(u *C.user, strategy *C.char, o *C.query_options) *C.bool_result {
-	res, err := cToGoUser(u).DeleteCredentials(C.GoString(strategy), SetQueryOptions(o))
-	return goToCBoolResult(res, err)
+//export kuzzle_security_create_restricted_user
+func kuzzle_security_create_restricted_user(k *C.kuzzle, body *C.char, o *C.query_options) *C.json_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateRestrictedUser(C.GoString(body), SetQueryOptions(o))
+	return goToCJsonResult(res, err)
 }
 
-//export kuzzle_security_user_get_credentials_info
-func kuzzle_security_user_get_credentials_info(u *C.user, strategy *C.char, o *C.query_options) *C.json_result {
-	res, err := cToGoUser(u).GetCredentialsInfo(C.GoString(strategy), SetQueryOptions(o))
+//export kuzzle_security_create_first_admin
+func kuzzle_security_create_first_admin(k *C.kuzzle, body *C.char, o *C.query_options) *C.json_result {
```

Comment:
We should not use json_result anymore but string_result as we use char* for json now